src/creds.rs

```diff
@@ -259,3 +193,48 @@ fn test_protocol_version() {
     assert_eq!(mls10_e[0], 1);
     assert_eq!(default_e[0], 1);
 }
+
+/// This struct contains a credential and the corresponding private key.
+pub struct CredentialBundle {
+    credential: Credential,
+    signature_private_key: SignaturePrivateKey,
+}
+
+impl CredentialBundle {
+    /// Create a new `CredentialBundle` of the given credential type for the
+    /// given identity and ciphersuite. The corresponding `SignatureKeyPair` is
+    /// freshly generated.
+    pub fn new(
+        identity: Vec<u8>,
+        credential_type: CredentialType,
+        ciphersuite_name: CiphersuiteName,
+    ) -> Result<Self, CredentialError> {
+        let keypair = Ciphersuite::new(ciphersuite_name).new_signature_keypair();
+        let ciphersuite = Ciphersuite::new(ciphersuite_name);
+        let mls_credential = match credential_type {
+            CredentialType::Basic => Ok(BasicCredential {
+                identity,
+                ciphersuite,
+                public_key: keypair.get_public_key().clone(),
+            }),
+            _ => Err(CredentialError::UnsupportedCredentialType),
+        }?;
+        let credential = Credential {
+            credential_type,
+            credential: MLSCredentialType::Basic(mls_credential),
+        };
+        Ok(CredentialBundle {
+            credential,
+            signature_private_key: keypair.get_private_key().clone(),
+        })
+    }
+
+    /// Get a reference to the signature private key.
+    pub fn signature_private_key(&self) -> &SignaturePrivateKey {
```

Comment:
Shouldn't this function go away and the `CredentialBundle` be passed around instead? I really don't like having to pull out a private key to do something.